tsMuxer/bufferedReaderManager.cpp

```diff
@@ -27,14 +27,7 @@ void BufferedReaderManager::init(uint32_t blockSize, uint32_t allocSize, uint32_
     m_prereadThreshold = prereadThreshold > 0 ? prereadThreshold : m_blockSize / 2;
 }
 
-BufferedReaderManager::~BufferedReaderManager()
-{
-    for (size_t i = 0; i < m_fileReaders.size(); i++)
-    {
-        delete m_fileReaders[i];  // need to define destruction order first. This object MUST be deleted after
-                                  // MCVodStreamer
-    }
-}
+BufferedReaderManager::~BufferedReaderManager() { std::vector<BufferedReader*>().swap(m_fileReaders); }
```

Comment:
This will cause memory leaks, since the `BufferedReader` objects inside `m_fileReaders` will never be destructed, as the vector consists of raw pointers, not `unique_ptr`s or similar objects which also manage the object's lifetime.
Since it's not straightforward to just change `BufferedReader*` to `unique_ptr<BufferedReader>` (which would certainly be welcome but it's definitely out of the scope of this PR), I'm afraid this loop must stay here.

Reply:
Ok, reverted.